site-manage/site-group-manager/src/main/java/org/sakaiproject/groupmanager/controller/AutoGroupsController.java

```diff
@@ -296,6 +298,25 @@ public String showStep4(Model model, @ModelAttribute AutoGroupsForm autoGroupsFo
                     }
                 }
             }
+
+            if (useManuallyAddedUsers) {
+                List<Member> allSiteMembers = site.getMembers().stream().filter(m -> selectedRoleList.contains(m.getRole().getId())).collect(Collectors.toList());
```

Comment:
I think I've written code in the past that gets the non-provided users in a site, which IIRC is less complex than this. I'll try and find it and post back here.